app/helpers/heatmap_helper.rb

```diff
@@ -81,9 +81,11 @@ def self.sample_taxons_dict(params, samples, background_id)
     taxon_ids = details.pluck('tax_id')
     taxon_ids -= removed_taxon_ids
 
-    # Refetch at genus level using species level
-    parent_ids = species_selected ? [] : HeatmapHelper.fetch_parent_ids(taxon_ids, samples)
-    results_by_pr = HeatmapHelper.fetch_samples_taxons_counts(samples, taxon_ids, parent_ids, background_id)
+    unless taxon_ids.empty?
+      # Refetch at genus level using species level
+      parent_ids = species_selected ? [] : HeatmapHelper.fetch_parent_ids(taxon_ids, samples)
+      results_by_pr = HeatmapHelper.fetch_samples_taxons_counts(samples, taxon_ids, parent_ids, background_id)
+    end
 
     HeatmapHelper.samples_taxons_details(
```

Comment:
Shouldn't we skip this too?

Reply:
For weird reasons, you still need to return the samples in the response, even if they don't have any results. I think SamplesHeatmapView clobbers this.state.sampleIds every time the response comes back. If you return {} in this endpoint, the heatmap page will forget about all sampleIds and you won't be able to get them back even if you remove the threshold filter...

Looks like a nontrivial refactor.